RFS/src/main/java/org/opensearch/migrations/bulkload/common/DocumentReindexer.java

```diff
@@ -53,6 +53,16 @@ Mono<Void> reindexDocsInParallelBatches(Flux<BulkDocSection> docs, String indexN
             .then();
     }
 
+    @SneakyThrows
+    BulkDocSection transformDocument(RfsLuceneDocument doc, String indexName) {
+        var original = new BulkDocSection(doc.id, indexName, doc.type, doc.source);
+        if (transformer != null) {
```

Comment:
Thinking in-text, how will a user know if a transformation was applied or not applied?  I'm thinking that this default transformation that removes types, it might be useful to know if you hit this scenario or not because it means your client requests must be altered.  This would impact some indices and not others.

With the current model, I suppose predicate can know if we ran or didn't run the transformation, but there are different transformations on indexes.

Reply:
I think both of these concerns are best solved from within the transformation as opposed to this code, we can add a metrics to the predicate transforms as we see fit. 

I've created [MIGRATIONS-2193](https://opensearch.atlassian.net/browse/MIGRATIONS-2193) to track this work